From 9f3df42bc2d07b607ffa2c538bc26cfd6ff775c2 Mon Sep 17 00:00:00 2001
From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Date: Tue, 6 Oct 2020 10:42:48 +0200
Subject: [PATCH 24/28] printk: Remove vprintk_default() and add printf
 annotation

commit e04642b8c492504251371aa316474e91f67430cc from linux-rt-devel

The kernel test robot reported an used function a missing printf
annotation for another function.
These changes were folded into the printk patch.

Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Signed-off-by: Ovidiu Panait <ovidiu.panait@windriver.com>
---
 kernel/printk/printk.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/kernel/printk/printk.c b/kernel/printk/printk.c
index f6cb243193b7..86023c9a22e0 100644
--- a/kernel/printk/printk.c
+++ b/kernel/printk/printk.c
@@ -1981,12 +1981,6 @@ asmlinkage int vprintk(const char *fmt, va_list args)
 }
 EXPORT_SYMBOL(vprintk);
 
-int vprintk_default(const char *fmt, va_list args)
-{
-	return vprintk_emit(0, LOGLEVEL_DEFAULT, NULL, 0, fmt, args);
-}
-EXPORT_SYMBOL_GPL(vprintk_default);
-
 /**
  * printk - print a kernel message
  * @fmt: format string
@@ -2782,7 +2776,7 @@ static int __init init_printk_kthread(void)
 }
 late_initcall(init_printk_kthread);
 
-static int vprintk_deferred(const char *fmt, va_list args)
+__printf(1, 0) static int vprintk_deferred(const char *fmt, va_list args)
 {
 	return vprintk_emit(0, LOGLEVEL_DEFAULT, NULL, 0, fmt, args);
 }
-- 
2.31.1

